sklearn/preprocessing/_encoders.py

```diff
@@ -430,6 +433,23 @@ class OneHotEncoder(_BaseEncoder):
            [1., 0., 0.]])
     """
 
+    _parameter_constraints = {
+        "categories": [StrOptions({"auto"}), "array-like"],
+        "drop": [StrOptions({"first", "if_binary"}), "array-like", None],
+        # By using object as parameter constraint, we are currently
+        # allowing numpy to do the validation. Numpy data types can take
+        # various inputs, that validate_params can't validate yet.
+        "dtype": [object],
```

Comment:
With https://github.com/scikit-learn/scikit-learn/pull/23602, this can be:

```suggestion
        "dtype": ["no_validation"],
```

to disable validation.

(You may need to sync with `main` to get the commit from #23602)